students/fahad/Lab 1/Part_4/Part_4.ino

```diff
@@ -0,0 +1,211 @@
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, 0);
+      delay(30);
+    }
+    for (int i = 0; i <= 255; i += 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, 0);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+    for (int i = 255; i >= 0; i -= 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, 0);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+    for (int i = 0; i <= 255; i += 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, 0);
+      delay(30);
+    }
+
+  }
+
+  // Fading RGB
+  else if (mode == 4)
+  {
+    for (int i = 255; i >= 0; i -= 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, 0);
+      delay(30);
+    }
+    for (int i = 0; i <= 255; i += 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, 0);
+      analogWrite(GLED, i);
+      analogWrite(BLED, 0);
+      delay(30);
+    }
+    for (int i = 255; i >= 0; i -= 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, 0);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+    for (int i = 0; i <= 255; i += 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, i);
+      analogWrite(BLED, 0);
+      delay(30);
+    }
+    for (int i = 255; i >= 0; i -= 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, 0);
+      analogWrite(GLED, i);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+    for (int i = 0; i <= 255; i += 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, 0);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+    for (int i = 255; i >= 0; i -= 5) {
+      if (debounce(lastButton) == true) {
+        break;
+      }
+      analogWrite(RLED, i);
+      analogWrite(GLED, i);
+      analogWrite(BLED, i);
+      delay(30);
+    }
+
+
+  }
+
+  //OFF (mode = 0)
+  else
+  {
+    digitalWrite(RLED, LOW);
+    digitalWrite(GLED, LOW);
+    digitalWrite(BLED, LOW);
+  }
+}
+void loop()
+{
+  currentButton = debounce(lastButton); // Read debounced state
+  if (lastButton == LOW && currentButton == HIGH) // If it was pressed...
+  {
+    ledMode++; // Increment the LED value
+  }
+  lastButton = currentButton; // Reset button value
+  // If you've cycled through the different options,
+  // reset the counter to 0
+  if (ledMode == 5) ledMode = 0;
```

Comment:
I would suggest putting all of this into a separate function, then calling that function when you check for the button inside the loops. Otherwise, you're doing the debounce, breaking, then debouncing again.